src/emulator/gui/src/game_selector.cpp

```diff
@@ -179,22 +190,105 @@ void draw_game_selector(GuiState &gui, HostState &host) {
                     }
                     if (ImGui::IsMouseClicked(0))
                         selected[0] = true;
+                    if (gui.title_id != game.title_id)
+                        gui.title_id = game.title_id;
+                }
+
+                const auto games_index = std::find_if(gui.game_selector.games.begin(), gui.game_selector.games.end(), [&game](const Game &g) {
+                    return g.app_ver == game.app_ver && g.title == game.title && g.title_id == game.title_id;
+                });
+
+                if (!fs::exists(game_path / game.title_id)) {
+                    gui.game_selector.games.erase(games_index);
+                    if (gui.delete_game_popup)
+                        gui.delete_game_popup = false;
+                }
+
+                if (gui.title_id == game.title_id) {
```

Comment:
My brain says this is always true. Double check if you need this `if` statement anymore.

Reply:
Is not all time true, already explain is obligatory for this function